pkg/java/pom/parse.go

```diff
@@ -164,16 +196,47 @@ func (p *parser) parseRoot(root artifact) ([]types.Library, []types.Dependency,
 	}
 
 	// Convert to []types.Library
-	for name, ver := range uniqArtifacts {
```

Comment:
you are using `processed` instead of `uniqArtifacts` here. I would suggest replace  `uniqArtifacts` in other places as well.
Keeping both collection may cause confusion in the future